docs/2_3_common_states.adoc

```diff
@@ -82,485 +87,644 @@ If `loggingOn == fmi3True`, the FMU enables a vendor defined set of `<LogCategor
-Such an event is called <<state event>>.
-_[This definition is slightly different from the standard definition of <<state event,`state events`>>: _ latexmath:[z_j(t) \cdot z_j(t_{i-1}) \leq 0] _._
-_This often used definition has the severe drawback that_ latexmath:[z_j(t_{i-1}) \neq 0] _is required in order to be well-defined and this condition cannot be guaranteed.]._
-All event indicators are piecewise continuous and are collected together in one vector of floating point numbers latexmath:[\mathbf{z(t)}]. +
+===== State: Event Mode [[EventMode,*Event Mode*]]
 
-.An event occurs when the event indicator changes its domain from latexmath:[z>0] to latexmath:[z\leq 0] or vice versa.
-[#figure-events]
-image::images/Event.svg[width=60%, align="center"]
+<<EventMode>> is only available in Model Exchange and Co-Simulation.
+In <<EventMode>> all continuous-time, discrete-time equations and active model partitions are evaluated.
 
-[start=4]
-. At every completed step of an integrator, <<fmi3CompletedIntegratorStep>> must be called (provided the capability flag `completedIntegratorStepNotNeeded` of `<fmiModelDescription>` is `false`).
-An event occurs at this time instant, if indicated by the return argument `enterEventMode == fmi3True`.
-Such an event is called <<step event>>.
-_[<<step event,`Step events`>> are, for example, used to dynamically change the (continuous) <<state,`states`>> of a model internally in the FMU, because the previous states are no longer suited numerically.]_
+There are multiple kinds of events that require a transition to <<EventMode>>:
+
+* <<time-event,Time events>>: The time instant of such an event is signaled to the importer by
+  - the argument <<nextEventTime>> of <<fmi3UpdateDiscreteStates>>, or
+  - the arguments of <<fmi3GetShift>> and <<fmi3GetInterval>>.
+* <<state-event,State events>>
+* <<input-event,Input events>>
+* <<step-event,Step events>>
+* Co-Simulation: <<fmi3DoStep>> returns with <<eventEncountered,`eventEncountered == fmi3True`>> or <<clocksAboutToTick,`clocksAboutToTick == fmi3True`>>.
+
+[#table-math-event-mode]
+[cols="5,3",options="header",]
+|====
+|Equations and Actions
+|Functions Influencing Equations
+
+|Set <<tunable>> <<parameter,`parameters`>> latexmath:[\mathbf{p}_{\mathit{variability == tunable}}]
+|`fmi3Set{VariableType}`
+
+|Set continuous-time and discrete-time <<input,`inputs`>> latexmath:[\mathbf{u}_{c+d}(\mathbf{t})]
+|`fmi3Set{VariableType}`
 
-In this mode all continuous-time and discrete-time equations are active and the unknowns at an event can be computed and retrieved.
-The event time of a <<state event>> may be determined if a domain change of at least one event indicator is detected at the end of a completed integrator step.
+a|
+* Activate clocks and respective model partitions in latexmath:[\mathbf{f}_{\mathit{event}}]
+* Deactivate clocks and
```

Comment:
I do not get it. Why deactivate clocks here?

Reply:
Because clocks might be conditional on some inputs. During numeric iteration in one super-dense time instant, this might activate and deactivate a clock. If not clear, please suggest concrete change.

Reply:
(My question was related to interpreting this table as describing a calling order. Then it would not have made sense to deactivate clocks at this point of time)